pkg/parser/expression_test.go

```diff
@@ -300,3 +304,76 @@ func TestParseTopLevelExpressionErr(t *testing.T) {
 		assert.Equal(t, wantErr, got.Error(), "input: %s\nerrors:\n%s", input, parser.errors)
 	}
 }
+
+func TestParseNiladic(t *testing.T) {
+	inputs := map[string]string{
+		"rand1":       "print rand1",
+		"rand1-twice": "print rand1 rand1",
+		"rand1-expr":  "print rand1+10",
+		"rand1-toplevelexpr": `
+n := rand1+0.5
+print n
+`,
+		"rand1-toplevelexpr-space": `
+if rand1 > 0.5
+  print "big"
+end
+`,
+		"custom1": `
+func answer:string
+	return "42"
+end
+
+print answer`,
+	}
+
+	for name, input := range inputs {
+		t.Run(name, func(t *testing.T) {
+			parser := newParser(input, testBuiltins())
+			_ = parser.parse()
+			assertNoParseError(t, parser, input)
+		})
+	}
+}
+
+type inputWithError struct {
```

Comment:
This type could move into the `ParseNiladicErr()` function. It could even be anonymous there too, but up to you.

Reply:
ok, i've done that. i usually don't do this either - along with my personal no nested funcs / types preference and a preference for shorter funcs. i know this is one of the few things we disagree on...